azext_edge/edge/providers/check/base.py

```diff
@@ -86,19 +87,24 @@ def filter_resources_by_name(
     resources: List[dict],
     resource_name: str,
 ) -> List[dict]:
-
-    if resource_name:
-        resource_name = resource_name.lower()
-        # check resource for wildcarded name
-        if "*" in resource_name:
-            regex = re.compile(f"^{resource_name.replace('*', '.*')}$")
-            resources = [resource for resource in resources if regex.match(
-                get_resource_metadata_property(resource, prop_name="name").lower()
-            )]
-        else:
-            resources = [resource for resource in resources if resource_name == get_resource_metadata_property(
-                resource, prop_name="name").lower()
-            ]
+    if not resource_name:
+        return resources
+    
+    # validate resource_name that should only contain alphanumeric characters, hyphens, ? and *
+    if not re.fullmatch(r"[a-zA-Z0-9\-?*]+", resource_name):
+        raise InvalidArgumentValueError("Invalid resource name. Only alphanumeric characters, hyphens, ? and * are allowed.")
+
+    resource_name = resource_name.lower()
+    # if only alphanumeric characters and hyphens, check resource for exact name
+    if "?" not in resource_name and "*" not in resource_name:
+        resources = [resource for resource in resources if resource_name == get_resource_metadata_property(
+            resource, prop_name="name").lower()
+        ]
+    else:
+        regex = re.compile(f"{resource_name.replace('*', '.*')}")
+        resources = [resource for resource in resources if regex.fullmatch(
+            get_resource_metadata_property(resource, prop_name="name").lower()
+        )]
```

Comment:
If we're already ensuring that the filter name does *not* include anything except alphanumeric, `*`, and `?` - we don't need to split the cases based on whether they have a glob pattern in them or not.

We can use the [`fnmatch`](https://docs.python.org/3/library/fnmatch.html) module to match glob-style strings to run `fnmatch(name, pattern)` for case-insensitive matches, or `fnmatchcase` for case-sensitive matching:

```
from fnmatch import fnmatch
resources = [
    resource
    for resource in resources
    if fnmatch(get_resource_metadata_property(resource, prop_name="name"), resource_name)
]
```

Reply:
got it, will test it out!